app/App.tsx

```diff
@@ -82,7 +82,7 @@ export const App = () => {
       ReactGA.set({ page });
       ReactGA.pageview(page);
     });
-  }, []);
+  }, [history]);
```

Comment:
This is unrelated to your PR, but I'm wondering if the `getLocation` call should be run within a `useMemo` hook. It's my understanding that the `useMemo` hook fires regardless of render status, so it would return the location data faster. If this sounds good, I will work on changing it in a future PR

Reply:
According to the [React docs](https://reactjs.org/docs/hooks-reference.html#usememo), `useMemo()` should only be used for memoizing expensive computations, and you shouldn't use it to do anything that shouldn't happen during rendering, such as side effects. Since our `getLocation()` function makes an API call to the browser location API and then falls back to making an API call to Google's location API, this is side effectful, so I don't think we want to use `useMemo()` for this. The React docs also say that `useMemo()` is not guaranteed to memoize all previous values, so it may have to rerun the function again, even if the dependencies haven't changed. This would be bad for side effectful things things like querying the browser location or firing off an API request to an external service.

I think if we wanted to fire off the location query before the initial render of the app, we might be able to do it by having a wrapper component whose only job is to fire off the location query, where we keep around a state variable that prevents the rest of the app from actually rendering until the effect has fired first. Something like:

```jsx
const LocationWrapper = () => {
  const [isFirstRender, setIsFirstRender] = useState(true);
  const [userLocation, setUserLocation] = useState(null);
  useEffect(() => {
    getLocation().then((x) => setUserLocation(x));
    setIsFirstRender(false);
  });

  if (isFirstRender) return null;
  return <App />;
}